store/store_helper.go

```diff
@@ -0,0 +1,63 @@
+package store
+
+import (
+	types "github.com/cosmos/cosmos-sdk/store/types"
+)
+
+func GetAndDecode[T any](store types.KVStore, dec func([]byte) (T, error), key []byte) (T, error) {
```

Comment:
These can be on the struct now :)

Reply:
No, because methods cannot have type parameters
See here: https://go.googlesource.com/proposal/+/refs/heads/master/design/43651-type-parameters.md#No-parameterized-methods